internal/optimization_test.go

```diff
@@ -0,0 +1,90 @@
+package internal
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDownloadOptimization(t *testing.T) {
+	// Setup test server
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte("test content"))
+	}))
+	defer ts.Close()
+
+	tmpDir := t.TempDir()
+	lockDir := filepath.Join(tmpDir, "001")
+	err := os.MkdirAll(lockDir, 0755)
+	require.NoError(t, err)
+
+	testUrl := ts.URL + "/test.txt"
+	testContent := []byte("test content")
+	testFile := filepath.Join(tmpDir, "test.txt")
+	err = os.WriteFile(testFile, testContent, 0644)
+	require.NoError(t, err)
+
+	t.Run("Valid File Not Redownloaded", func(t *testing.T) {
+		lockPath := filepath.Join(lockDir, "grabit.lock")
+		lock, err := NewLock(lockPath, true)
+		require.NoError(t, err)
+
+		err = lock.AddResource([]string{testUrl}, RecommendedAlgo, nil, "test.txt")
+		require.NoError(t, err)
+
+		err = lock.Download(tmpDir, nil, nil, "")
+		require.NoError(t, err)
+	})
+
+	t.Run("Invalid File Redownloaded", func(t *testing.T) {
+		lockPath := filepath.Join(lockDir, "grabit.lock")
+		lock, err := NewLock(lockPath, true)
+		require.NoError(t, err)
+
+		err = lock.AddResource([]string{testUrl}, RecommendedAlgo, nil, "test.txt")
```

Comment:
I see that all your tests use the same file... isn't this a bit dangerous in terms of stability/correctness?